bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -334,7 +334,8 @@ private void decodeProperties(final BsonReader reader, final DecoderContext deco
 
         for (Map.Entry<String, TypeParameterMap> entry : classModel.getPropertyNameToTypeParameterMap().entrySet()) {
             TypeParameterMap typeParameterMap = entry.getValue();
-            if (typeParameterMap.hasTypeParameters() && classModel.getPropertyModel(entry.getKey()).getCodec() == null) {
+            PropertyModel<?> propertyModel = classModel.getPropertyModel(entry.getKey());
+            if (typeParameterMap.hasTypeParameters() && propertyModel != null && propertyModel.getCodec() == null) {
```

Comment:
I was expecting

   ... && (propertyModel == null || propertyModel.getCodec() == null)

Can you just describe the intent here, because I'm not sure I get it.

Reply:
Thats probably more correct, the issue is we have a typeParameterMap but no propertyModel. Thanks to the peculiarities of ENUMs.  Alternative patch is up.